### ISSUE TITLE
Why dp-sgd working so slow?

### ISSUE THREAD

**Issue by inyutin** [tensorflow/privacy#109]
First of all, thank you for all of your work!

I try to understand how dp-sgd work. I run tutorial.mnist_dpsgd_tutorial.py and can't understand why it start to work so slow.
In naive SGD in work quite fast, 100 steps takes about 10 seconds. However, dp-sgd on the same CNN takes 180 seconds on 100 steps. 

As I see, dp-sgd add two more operations to naive sgd: gradient clipping and adding noise. All those operations should not be computational bound operations. Am I wrong?

I have no idea, why it dp-sgd slowdowns training process. I'd appreciate if you could give me a hint.

**Comment by inyutin** [comment]
I missed out the num_microbatches parameter :)
